aws/data_aws_crossaccount_policy.go

```diff
@@ -3,149 +3,295 @@ package aws
+				"ec2:DetachVolume",
+				"ec2:DisassociateIamInstanceProfile",
+				"ec2:ReplaceIamInstanceProfileAssociation",
+				"ec2:RevokeSecurityGroupEgress",
+				"ec2:RevokeSecurityGroupIngress",
+				"ec2:RunInstances",
+				"ec2:TerminateInstances",
+			}...)
+		}
+		// additional permissions for Databricks-managed VPC policy
+		if data.PolicyType == "managed" {
+			actions = append(actions, []string{
+				"ec2:CreateDhcpOptions",
+				"ec2:CreateInternetGateway",
+				"ec2:CreateNatGateway",
+				"ec2:CreateRoute",
+				"ec2:CreateRouteTable",
+				"ec2:CreateSecurityGroup",
+				"ec2:CreateSubnet",
+				"ec2:CreateVpc",
+				"ec2:CreateVpcEndpoint",
+				"ec2:DeleteDhcpOptions",
+				"ec2:DeleteInternetGateway",
+				"ec2:DeleteNatGateway",
+				"ec2:DeleteRoute",
+				"ec2:DeleteRouteTable",
+				"ec2:DeleteSecurityGroup",
+				"ec2:DeleteSubnet",
+				"ec2:DeleteVpc",
+				"ec2:DeleteVpcEndpoints",
+				"ec2:DetachInternetGateway",
+				"ec2:DisassociateRouteTable",
+				"ec2:ModifyVpcAttribute",
+				"ec2:ReleaseAddress",
+			}...)
+		}
+		policy := awsIamPolicy{
+			Version: "2012-10-17",
+			Statements: []*awsIamPolicyStatement{
+				{
+					Effect:    "Allow",
+					Actions:   actions,
+					Resources: "*",
+				},
+				{
+					Effect: "Allow",
+					Actions: []string{
+						"iam:CreateServiceLinkedRole",
+						"iam:PutRolePolicy",
 					},
-					{
-						Effect: "Allow",
-						Actions: []string{
-							"iam:CreateServiceLinkedRole",
-							"iam:PutRolePolicy",
-						},
-						Resources: "arn:aws:iam::*:role/aws-service-role/spot.amazonaws.com/AWSServiceRoleForEC2Spot",
-						Condition: map[string]map[string]string{
-							"StringLike": {
-								"iam:AWSServiceName": "spot.amazonaws.com",
-							},
+					Resources: "arn:aws:iam::*:role/aws-service-role/spot.amazonaws.com/AWSServiceRoleForEC2Spot",
+					Condition: map[string]map[string]string{
+						"StringLike": {
+							"iam:AWSServiceName": "spot.amazonaws.com",
 						},
 					},
 				},
-			}
-			if passRoleARNs, ok := d.GetOk("pass_roles"); ok {
-				policy.Statements = append(policy.Statements, &awsIamPolicyStatement{
+			},
+		}
+		// pass role permissions
+		if len(data.PassRole) > 0 {
+			policy.Statements = append(policy.Statements,
+				&awsIamPolicyStatement{
 					Effect:    "Allow",
 					Actions:   "iam:PassRole",
-					Resources: passRoleARNs,
+					Resources: data.PassRole,
 				})
-			}
-			policyJSON, err := json.MarshalIndent(policy, "", "  ")
-			if err != nil {
-				return err
-			}
-			d.SetId("cross-account")
-			// nolint
-			d.Set("json", string(policyJSON))
-			return nil
-		},
-		Schema: map[string]*schema.Schema{
-			"pass_roles": {
-				Type: schema.TypeList,
-				Elem: &schema.Schema{
-					Type: schema.TypeString,
+		}
+
+		// resource-based permissions
+		if data.PolicyType == "restricted" {
+			region := data.Region
+			aws_account_id := data.AwsAccountId
+			vpc_id := data.VpcId
+			security_group_id := data.SecurityGroupId
```

Comment:
Should we add validation for these parameters when `policy_type = restricted`? Either that they are non-empty, valid region/account/vpc ID/SG ID?

Reply:
added some validation